### PR TITLE
feat(sdk): initial checkin of golang nexus server

### PR DESCRIPTION
**Pull request by raubitsj**
Initial checkin of golang version of wandb nexus.

This is functional with many caveats.

This branch differs from https://github.com/wandb/nexus-golang/pull/1 in that:
- an updated golang structure
- most multi-language code has been removed (julia, C, wasm)

This still requires this wandb/wandb branch: https://github.com/wandb/wandb/pull/4419

To generate code:
```
# this will generate protobuf and graphql bindings
cp FROM_WANDB_CORE/schema.graphql api/graphql/
go generate
```

To use:
```
cd nexus-golang
pip install .
WANDB__USE_NEXUS=true python train.py
```

### CODE REVIEW

File: pkg/server/lib.go
Comment (kptkin) [inline pkg/server/lib.go:1]: should we move it out of server?

File: pkg/server/util.go
Comment (kptkin) [inline pkg/server/util.go:1]: probably need to move some of the logic into stream?
and maybe util should be of server?

File: pkg/service/wandb_base.pb.go
Comment (kptkin) [inline pkg/service/wandb_base.pb.go:1]: not sure where the auto gen files should go

**Review by kptkin — APPROVED** [review]
minor comments but we can iterate on this as we build it up

File: pkg/server/lib.go
Comment (raubitsj) [reply pkg/server/lib.go:1]: yes, likely want to move this as we create better abstractions for the lib interface

**Pull request merged**